mlflow/models/evaluation/base.py

```diff
@@ -425,9 +425,42 @@ def _hash_uint64_ndarray_as_bytes(array):
     return struct.pack(f">{array.size}Q", *array)
 
 
+def _is_array_has_dict(nd_array):
+    if len(nd_array) == 0:
+        return False
+
+    # It is less likely the array or list contains heterogeneous elements, so just checking the
+    # first element to avoid performance overhead.
+    elm = nd_array[0]
+    if isinstance(elm, (list, np.ndarray)):
+        return _is_array_has_dict(elm)
+    elif isinstance(elm, dict):
+        return True
+
+    return False
+
+
+def _hash_array_of_dict_as_bytes(data):
```

Comment:
note: This hashing support is required for passing a list of dictionary (chat/completion payload) or similar pandas DataFrame as a input dataset. The hashing is required for generating dataset digest, but it doesn't handle a list or array contains dictionary. The implementation is not super clean, just band-aiding without breaking existing use cases.